src/Engine/ProtoCore/Namespace/SymbolTable.cs

```diff
@@ -103,6 +242,28 @@ public override bool Equals(object obj)
         }
 
         /// <summary>
+        /// Equality operator overload to have same equality behaviour as object.Equals() override
+        /// </summary>
+        /// <param name="thisSymbol"></param>
+        /// <param name="otherSymbol"></param>
+        /// <returns></returns>
+        public static bool operator ==(Symbol thisSymbol, Symbol otherSymbol)
```

Comment:
No equality operator does not know about `object.Equals` as it is purely a C# thing and has not got anything to do with .NET, which is what defines `object.Equals`. Equality operator does not use `object.Equals` internally.